component/loki/source/kubernetes_events/event_controller.go

```diff
@@ -97,7 +97,9 @@ func (ctrl *eventController) runError(ctx context.Context) error {
 	}
 
 	defaultNamespaces := map[string]cache.Config{}
-	defaultNamespaces[ctrl.task.Namespace] = cache.Config{}
+	if ctrl.task.Namespace != "" {
```

Comment:
Should we _not_ initialise the map if the string is empty (and move the initialization inside this if clause?